ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/helpcenter/searcharticlepage/SearchArticleContainerFragmentTest.java

```diff
@@ -0,0 +1,223 @@
+    private EditText editText;
+
+    @Mock
+    private Editable editable;
+
+    @Mock
+    private SearchArticleContainerContract.Presenter presenter;
+
+    @Mock
+    private TextView textView;
+
+    @Mock
+    private KeyEvent keyEvent;
+
+    @Mock
+    private SearchArticleFragment searchArticleFragment;
+
+    @Mock
+    private MenuItem menuItem;
+
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptor;
+
+    @Before
+    public void setUp() {
+        searchArticleContainerFragment = (SearchArticleContainerFragment)SearchArticleContainerFragment.newInstance();
+    }
+
+    @Test
+    public void onCreate() {
+        //Act
+        searchArticleContainerFragment.onCreate(bundle);
+        final SearchArticleContainerContract.Presenter presenterLocal =
+                Whitebox.getInternalState(searchArticleContainerFragment, "mPresenter");
+
+        //Assert
+        assertNotNull(presenterLocal);
+    }
+
+    @Test
+    public void onCreateViewWhenAddTextChangedListener() throws NoSuchMethodException {
+        //Arrange
+        final int actionId = EditorInfo.IME_ACTION_SEARCH;
+        onCreateViewSupport();
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                Object arguments[] = invocationOnMock.getArguments();
+                final TextWatcher textWatcher = (TextWatcher)arguments[0];
+                textWatcher.beforeTextChanged(CHAR_SEQUENCE, i, i1, i2);
+                textWatcher.onTextChanged(CHAR_SEQUENCE, i, i1, i2);
+                textWatcher.afterTextChanged(editable);
+                return null;
+            }
+        }).when(editText).addTextChangedListener(any(TextWatcher.class));
+
+        //Act
+        searchArticleContainerFragment.onCreateView(layoutInflater, viewGroup, bundle);
+
+        //Assert
+        verify(presenter).onTextEntered(stringArgumentCaptor.capture());
+        assertEquals(CHAR_SEQUENCE.toString(), stringArgumentCaptor.getValue());
+    }
+
+    @Test
+    public void onCreateViewWhenOnEditorActionListener() throws NoSuchMethodException {
+        //Arrange
+        final int actionId = EditorInfo.IME_ACTION_SEARCH;
+        onCreateViewSupport();
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+                Object arguments[] = invocationOnMock.getArguments();
+                final TextView.OnEditorActionListener onEditorActionListener =
+                        (TextView.OnEditorActionListener)arguments[0];
+                onEditorActionListener.onEditorAction(textView, actionId, keyEvent);
+                return null;
+            }
+        }).when(editText).setOnEditorActionListener(any(TextView.OnEditorActionListener.class));
+
+        //Act
+        searchArticleContainerFragment.onCreateView(layoutInflater, viewGroup, bundle);
+
+        //Assert
```

Comment:
✅ Good to see that the AAA("Arrange-Act-Assert") arranging and formatting pattern is followed.

Some of the main advantages are that it clearly separates what is being tested from the setup and verification steps. It also clarifies and focuses attention on a historically successful and generally necessary set of test steps.

**No change suggested, looks good!**	  

<a id="73e74080-537e-11e8-b287-f5597e01cef6" href="none" codereview-uuid-reserved-tag />